src/ledger/LedgerManager.h

```diff
@@ -16,27 +16,141 @@ class LedgerCloseData;
+//        │    │   │    ┌──┼───────────────────────────┐
+//        │    │   │    │  │   LedgerApplyManager      │
+//        │    │   │    │  │                           │
+//        │    │   │    │  │  Assembles contiguous     │
+//      T │    │ M │    │  │  ledger sequence and      │
+//      i │    │ a │    │  │  feeds to apply thread    │
+//      m │    │ i │    │  │  (or triggers catchup)    │
+//      e │    │ n │    │  │                           │
+//        │    │   │    │  │ ┌───────────────────┐     │    ┌───┐
+//      p │    │ T │    │  ▼ │  mSyncingLedgers  │─────┼───►│   │
+//      a │    │ h │    │    └───────────────────┘     │    │   │
+//      s │    │ r │    │   L                     Q    │    │   │
+//      s │    │ e │    └──────────────────────────────┘    │   │
+//      i │    │ a │                                        │ A │
+//      n │    │ d │                                        │ p │
+//      g │    │   │    ┌──────────────────────┐            │ p │
+//        │    │   │    │                      │            │ l │
+//        │    │   │    │                      │            │ y │
+//        │    │   │    │                      │            │   │
+//        │    │   │    │    LedgerManager     │         A  │ T │
+//        │    │   │    │                      │            │ h │
+//        │    │   │    │ State and logic for  │            │ r │
+//        ▼    │   │    │ apply thread and     │            │ e │
+//             │   │    │ close, including     │            │ a │
+//             │   │    │ storing snapshot of  │            │ d │
+//             │   │    │ "last closed ledger" │            │   │
+//             │   │    │                      │            │   │
+//             │   │    │             LCL ◄────┼────────────┤   │
+//             │   │    │                      │            └──┬┘
+//             │   │    │                      │             ▲ ▼
+//             └───┘    │ mLastClosedLedger    │      ┌──────┴───┐
+//                      │ mLastClosedLedgerHAS │      │ DB       │
+//                      │ mLastClosedSnapshot ─┼─────►│ &        │
+//                      │                      │      │ Buckets  │
+//                      └──────────────────────┘      └──────────┘
+//
+//
+// The apply thread uses the buckets and database, as well as some state
+// variables in the LM. The remaining state variables of the LM, including the
+// LCL variables, are for the main thread. One of the LCL state variables is a
+// snapshot of the bucketlist. The bucketlist snapshots are immutable and
+// therefore trivially threadsafe, but may be lagging behind the newest
+// bucketlist formed by the apply thread.
+//
+// In more precise terms, 4 points on the diagram are labeled: L, Q, A, and LCL.
+// These are points where we can identify and relate some state variables and an
+// invariant order between them:
+//
+//     L -- LedgerApplyManagerImpl::mLargestLedgerSeqHeard tracks this, it is
```

Comment:
Nit: I think changing L -> H makes a little more sense to me. This disambiguates it more with LCL and also gives us:
H -> (largest) Heard
Q -> (largest) Queued
A -> Applied
LCL -> Last Closed (which should probably really be C, but LCL is maintained for legacy reasons).

Reply:
done